server/embed/config.go

```diff
@@ -1084,8 +1102,14 @@ func (cfg *Config) Validate() error {
 		return fmt.Errorf("setting experimental-enable-lease-checkpoint-persist requires experimental-enable-lease-checkpoint")
 	}
 
-	if cfg.ExperimentalCompactHashCheckTime <= 0 {
-		return fmt.Errorf("--experimental-compact-hash-check-time must be >0 (set to %v)", cfg.ExperimentalCompactHashCheckTime)
+	if cfg.ExperimentalCompactHashCheckTime < 0 {
+		return fmt.Errorf("--experimental-compact-hash-check-time must be >=0 (set to %v)", cfg.ExperimentalCompactHashCheckTime)
+	}
+	if cfg.ExperimentalCompactHashCheckTime > 0 {
+		cfg.CompactHashCheckTime = cfg.ExperimentalCompactHashCheckTime
```

Comment:
Should we do this in validate function?

Reply:
`Validate` shouldn't change any field's value. Suggest to do this post-validate, i.e. in `parse` (at line >= 167 see below), so that it can take care of both command line and config file cases.

https://github.com/etcd-io/etcd/blob/026bbecd25e0f9312b99175940a6761580cf5643/server/etcdmain/config.go#L125-L167